exonum-java-binding/core/rust/src/utils/conversion.rs

```diff
@@ -30,10 +31,38 @@ pub fn convert_hash(env: &JNIEnv, hash: &Hash) -> JniResult<jbyteArray> {
     env.byte_array_from_slice(hash.as_ref())
 }
 
-/// Converts JNI `JString` into Rust `String`
+/// Converts JNI `JString` into Rust `String`.
 pub fn convert_to_string<'e, V>(env: &JNIEnv<'e>, val: V) -> JniResult<String>
 where
     V: Into<JString<'e>>,
 {
     Ok(env.get_string(val.into())?.into())
 }
+
+/// Converts anything convertible to a protobuf message into Java byte array.
+pub fn proto_to_java_bytes(
+    env: &JNIEnv,
+    proto: impl ProtobufConvert<ProtoStruct = impl Message>,
+) -> JniResult<jbyteArray> {
+    let bytes = proto.to_pb().write_to_bytes().unwrap();
+    env.byte_array_from_slice(&bytes)
+}
+
+/// Converts array of Java bytes arrays to the vector of Rust array representation.
+pub fn java_arrays_to_rust<T, F>(
+    env: &JNIEnv,
+    array: jobjectArray,
+    to_rust_array: F,
+) -> JniResult<Vec<T>>
+where
+    F: Fn(&JNIEnv, jbyteArray) -> JniResult<T>,
+{
+    let num_elements = env.get_array_length(array)?;
+    let mut result = Vec::with_capacity(num_elements as usize);
+    for i in 0..num_elements {
```

Comment:
With this basically `.map().collect(toList())`, have we considered implementing an Iterator of auto_locals for `jobjectArray`s? I am not sure we shall (put that into the jni library) because it will become a source of leaks (i.e., any operation producing new locals — filter, map, etc. — shall manage them properly) unless the user is well aware of the responsibility.

Reply:
@skletsun , @vitvakatu ?

Reply:
I don't see the way how `filter` or `map` usage will lead to autolocals leaks 🤔

Reply:
If I get things right `filter`/`map`/`whatever` should not produce new local but operate with existing ones created by an iterator. And at first look, the user should only be aware that object disappears after iteration unless global reference was created.
The problem that I see right now is that implementing, for example, `IntoIterator` for `jobjectArray` means implementing it for any object because `jobjectArray` is not a separate type but just a type alias for `jarray` which in turn an alias for `jobject`.

Reply:
I thought of something like
```
var yellow_bananas = env.stream(array_of_bananas)
   .filter(banana_auto_local -> {
    // Stupid code *creating a new local for color*
    Color brownColor = env.get_field(Color.Brown);
    return !env.call_method(banana_auto_local, "hasColor", brownColor)
  })
  .collect(toVec());
```

Reply:
>
```

  .collect(toVec());
```

Also, array of `AutoLocal`s kind of contradits their purpose

Reply:
I think we shall keep this in mind, in case we could find a good (safe, ergonomic) way to implement that.